common/flock.py

```diff
@@ -84,23 +84,26 @@ def __init__(self,
         self._file_path = folder / filename
 
         if self._can_use_file(self._file_path):
-            return
+            return None
 
         # Try user specific file lock
         # e.g. /run/user/<UID>
-        self._file_path = Path(os.environ['XDG_RUNTIME_DIR']) / filename
+        self._file_path = Path(
+            os.environ.get('XDG_RUNTIME_DIR',
+                           f'/run/user/{os.getuid()}')
+        ) / filename
```

Comment:
I assume that changing `return` into `return None` is just a question of style, and has no functional effect?

Otherwise, this looks sensible, *if* we can assume that `/run/user/{os.getuid()}` is available on every distro these days.

Reply:
`return None` is the suggested alternative to `return` where a method has a return statement with a return value specified, both in my testing and the documentation for the pylint warning [here](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/inconsistent-return-statements.html). A quick test shows that methods that either don't specify return values or that do not use a return statement return `None` by default.

According to [the library reference](https://docs.python.org/3/library/os.html#os.getuid), `os.getuid()` is available in all Python versions on Unix-based systems, but is a stub on Emscripten and WASI.

Reply:
Yes __init__() always return None. Here we just make it explicit instead of implicit. Python Zen rule #2.

About `/run/user/{os.getuid()}`: Ever folder that is used as flock is tested first (see `_can_use_file()`). Otherwise the next alternative folder will be tried and in the end an exception should be raised.